File: 1291-sequential-digits/1291-sequential-digits.py
class Solution:
    def sequentialDigits(self, low: int, high: int) -> List[int]:
        s = "".join([str(i) for i in range(1, 10)])
        res = []
        n = len(str(low))
        m = len(str(high))
        
        for p in range(n, m+1):
            for q in range(10-p):
                if low <= int(s[q:q+p]) <= high:
                    print(s[q:q+p])
                    res.append(int(s[q:q+p]))
        return res

File: 1291-sequential-digits/1291-sequential-digits.java
class Solution {
    public List<Integer> sequentialDigits(int low, int high) {
        String s = "123456789";
        List<Integer> res = new ArrayList<>();
        
        int n = String.valueOf(low).length();
        int m = String.valueOf(high).length();
        
        for (int i = n; i <= m; i++) {
            for (int j = 0; j < 10-i; j++) {
                int val = Integer.parseInt(s.substring(j, j+i));
                if (low <= val && val <= high) {
                    res.add(val);
                }
            }
        }
        return res;
    }
}